### ISSUE TITLE
navigator.id.get actually logs user in if dialog is dismissed

### ISSUE THREAD

**Issue by jaredhirsch** [mozilla/persona#2437]
@jbuck dropped by #identity and reported the bug; I was able to reproduce in latest Chrome and FF Nightly.

If you have logged in before, then go back to the page, start the login flow, but close the popup, the login callback gets called twice--once with a null assertion, then the second time, with an assertion. So the user winds up getting logged back in.

STR:
1. go to dev.butyr.org/templates/basic, log in using persona
2. log out by clicking the email address
3. reload page
4. click 'sign in'
5. close the persona dialog, don't complete the sign in process
6. the Cornfield.login function (around line 9738 in butter.js) will get called twice, with an assertion the second time
7. user gets logged back in

**Comment by jaredhirsch** [comment]
@jbuck Looks like butyr.org is gone, any way that we can try to repro this?

**Comment by jbuck** [comment]
I tried to recreate the state of dev.butyr.org at that time by doing the following:
1. `git clone https://github.com/mozilla/butter.git`
2. `cd butter`
3. `git checkout v0.8`
4. `npm install`
5. `git submodule update --init`
6. `node make server`

However, I'm getting some errors in the butter that I have no idea how to solve... things like the session is not sticking around. We switched to the .watch() API ourselves, maybe just close this issue?
